composition-js/src/__tests__/cachetag.test.ts

```diff
@@ -0,0 +1,364 @@
+import { composeServices } from "../compose";
+import { printSchema } from "@apollo/federation-internals";
+import { parse } from "graphql/index";
+
+describe("cacheTag spec and join__directive", () => {
+  it("composes", () => {
+    const subgraphs = [
+      {
+        name: "products",
+        typeDefs: parse(`
+                    extend schema
+                    @link(
+                        url: "https://specs.apollo.dev/federation/v2.12"
+                        import: ["@key" "@cacheTag"]
+                    )
+
+                    type Query {
+                        resources: [Resource!]! @cacheTag(format: "resources")
+                    }
+
+                    type Resource @key(fields: "id") @cacheTag(format: "resource-{$key.id}") {
+                        id: ID!
+                        name: String!
+                    }
+                `),
+      },
+    ];
+
+    const result = composeServices(subgraphs);
+    expect(result.errors ?? []).toEqual([]);
+    const printed = printSchema(result.schema!);
+    expect(printed).toMatchInlineSnapshot(`
+      "schema
+        @link(url: \\"https://specs.apollo.dev/link/v1.0\\")
+        @link(url: \\"https://specs.apollo.dev/join/v0.5\\", for: EXECUTION)
+        @link(url: \\"https://specs.apollo.dev/cacheTag/v0.1\\", for: EXECUTION)
+      {
+        query: Query
+      }
+
+      directive @link(url: String, as: String, for: link__Purpose, import: [link__Import]) repeatable on SCHEMA
+
+      directive @join__graph(name: String!, url: String!) on ENUM_VALUE
+
+      directive @join__type(graph: join__Graph!, key: join__FieldSet, extension: Boolean! = false, resolvable: Boolean! = true, isInterfaceObject: Boolean! = false) repeatable on OBJECT | INTERFACE | UNION | ENUM | INPUT_OBJECT | SCALAR
+
+      directive @join__field(graph: join__Graph, requires: join__FieldSet, provides: join__FieldSet, type: String, external: Boolean, override: String, usedOverridden: Boolean, overrideLabel: String, contextArguments: [join__ContextArgument!]) repeatable on FIELD_DEFINITION | INPUT_FIELD_DEFINITION
+
+      directive @join__implements(graph: join__Graph!, interface: String!) repeatable on OBJECT | INTERFACE
+
+      directive @join__unionMember(graph: join__Graph!, member: String!) repeatable on UNION
+
+      directive @join__enumValue(graph: join__Graph!) repeatable on ENUM_VALUE
+
+      directive @join__directive(graphs: [join__Graph!], name: String!, args: join__DirectiveArguments) repeatable on SCHEMA | OBJECT | INTERFACE | FIELD_DEFINITION
+
+      directive @cacheTag(format: String!) repeatable on OBJECT | INTERFACE | FIELD_DEFINITION
```

Comment:
I thought we only wanted this as a marker spec in the supergraph format and not a new directive?

Reply:
No, it's a unfortunate side effect of adding the spec. I'll find a way to drop it.